tests/by-util/test_numfmt.rs

```diff
@@ -145,3 +145,114 @@ fn test_si_to_iec() {
         .run()
         .stdout_is("13.9T\n");
 }
+
+#[test]
+fn test_should_report_invalid_empty_number_on_empty_stdin() {
+    new_ucmd!()
+        .args(&["--from=auto"])
+        .pipe_in("\n")
+        .run()
+        .stderr_is("numfmt: invalid number: ‘’\n");
+}
+
+#[test]
+fn test_should_report_invalid_empty_number_on_blank_stdin() {
+    new_ucmd!()
+        .args(&["--from=auto"])
+        .pipe_in("  \t  \n")
+        .run()
+        .stderr_is("numfmt: invalid number: ‘’\n");
+}
+
+#[test]
+fn test_should_report_invalid_suffix_on_stdin() {
+    new_ucmd!()
+        .args(&["--from=auto"])
+        .pipe_in("1k")
+        .run()
+        .stderr_is("numfmt: invalid suffix in input: ‘1k’\n");
+
+    // GNU numfmt reports this one as “invalid number”
+    new_ucmd!()
+        .args(&["--from=auto"])
+        .pipe_in("NaN")
+        .run()
+        .stderr_is("numfmt: invalid suffix in input: ‘NaN’\n");
+}
+
+#[test]
+fn test_should_report_invalid_number_with_interior_junk() {
+    // GNU numfmt reports this as “invalid suffix”
+    new_ucmd!()
+        .args(&["--from=auto"])
+        .pipe_in("1x0K")
+        .run()
+        .stderr_is("numfmt: invalid number: ‘1x0K’\n");
+}
+
+#[test]
+fn test_should_skip_leading_space_from_stdin() {
+    new_ucmd!()
+        .args(&["--from=auto"])
+        .pipe_in(" 2Ki")
+        .run()
+        .stdout_is("2048\n");
```

Comment:
@sylvestre we've got some cases around handling leading padding, trailing data, & multiple lines from stdin, some tests for handling padding in free arguments, and a couple checks of error messages on bad input.

What are some additional cases I should add?

thanks

Reply:
Making sure that your change is tested 
Or to rephrase, if we revert your change, the new test should broke.

Reply:
I see. Here is a test run with the new tests against `numfmt.rs` from `master`:

```
~/sys/coreutils [numfmt-leading-whitespace|✔]$ git checkout master -- src/uu/numfmt/src/numfmt.rs
~/sys/coreutils [numfmt-leading-whitespace|●1✚ 1]$ cargo test --features "numfmt" --no-default-features 2>/dev/null

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


running 26 tests
test test_numfmt::test_from_auto ... ok
test test_numfmt::test_from_si ... ok
test test_numfmt::test_from_iec_i ... ok
test test_numfmt::test_leading_whitespace_in_free_argument_should_imply_padding ... FAILED
test test_numfmt::test_header_default ... ok
test test_numfmt::test_header ... ok
test test_numfmt::test_input_from_free_arguments ... ok
test test_numfmt::test_from_iec ... ok
test test_numfmt::test_negative_padding ... ok
test test_numfmt::test_leading_whitespace_should_imply_padding ... FAILED
test test_numfmt::test_no_op ... ok
test test_numfmt::test_normalize ... ok
test test_numfmt::test_padding ... ok
test test_numfmt::test_should_calculate_implicit_padding_per_free_argument ... FAILED
test test_numfmt::test_should_calculate_implicit_padding_per_line ... FAILED
test test_numfmt::test_should_report_invalid_empty_number_on_empty_stdin ... FAILED
test test_numfmt::test_should_convert_only_first_number_in_line ... FAILED
test test_numfmt::test_should_report_invalid_number_with_interior_junk ... FAILED
test test_numfmt::test_should_report_invalid_empty_number_on_blank_stdin ... FAILED
test test_numfmt::test_should_report_invalid_suffix_on_stdin ... FAILED
test test_numfmt::test_negative ... ok
test test_numfmt::test_to_iec_i ... ok
test test_numfmt::test_should_skip_leading_space_from_stdin ... FAILED
test test_numfmt::test_to_iec ... ok
test test_numfmt::test_si_to_iec ... ok
test test_numfmt::test_to_si ... ok

failures:

---- test_numfmt::test_leading_whitespace_in_free_argument_should_imply_padding stdout ----
current_directory_resolved:
run: /home/drocco/sys/coreutils/target/debug/coreutils numfmt --from=auto    1Ki
thread 'test_numfmt::test_leading_whitespace_in_free_argument_should_imply_padding' panicked at 'assertion failed: `(left == right)`
  left: `""`,
 right: `"  1024\n"`', tests/common/util.rs:119:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_numfmt::test_leading_whitespace_should_imply_padding stdout ----
current_directory_resolved:
run: /home/drocco/sys/coreutils/target/debug/coreutils numfmt --from=auto
thread 'test_numfmt::test_leading_whitespace_should_imply_padding' panicked at 'assertion failed: `(left == right)`
  left: `""`,
 right: `" 1000\n"`', tests/common/util.rs:119:9

---- test_numfmt::test_should_calculate_implicit_padding_per_free_argument stdout ----
current_directory_resolved:
run: /home/drocco/sys/coreutils/target/debug/coreutils numfmt --from=auto    1Ki         2K
thread 'test_numfmt::test_should_calculate_implicit_padding_per_free_argument' panicked at 'assertion failed: `(left == right)`
  left: `""`,
 right: `"  1024\n      2000\n"`', tests/common/util.rs:119:9

---- test_numfmt::test_should_calculate_implicit_padding_per_line stdout ----
current_directory_resolved:
run: /home/drocco/sys/coreutils/target/debug/coreutils numfmt --from=auto
thread 'test_numfmt::test_should_calculate_implicit_padding_per_line' panicked at 'assertion failed: `(left == right)`
  left: `""`,
 right: `"  1024\n      2000\n"`', tests/common/util.rs:119:9

---- test_numfmt::test_should_report_invalid_empty_number_on_empty_stdin stdout ----
current_directory_resolved:
run: /home/drocco/sys/coreutils/target/debug/coreutils numfmt --from=auto
thread 'test_numfmt::test_should_report_invalid_empty_number_on_empty_stdin' panicked at 'assertion failed: `(left == right)`
  left: `"numfmt: error: Failed to parse suffix"`,
 right: `"numfmt: invalid number: ‘’"`', tests/common/util.rs:133:9

---- test_numfmt::test_should_convert_only_first_number_in_line stdout ----
current_directory_resolved:
run: /home/drocco/sys/coreutils/target/debug/coreutils numfmt --from=auto
thread 'test_numfmt::test_should_convert_only_first_number_in_line' panicked at 'assertion failed: `(left == right)`
  left: `""`,
 right: `"1024 2M 3G\n"`', tests/common/util.rs:119:9

---- test_numfmt::test_should_report_invalid_number_with_interior_junk stdout ----
current_directory_resolved:
run: /home/drocco/sys/coreutils/target/debug/coreutils numfmt --from=auto
thread 'test_numfmt::test_should_report_invalid_number_with_interior_junk' panicked at 'assertion failed: `(left == right)`
  left: `"numfmt: error: invalid float literal"`,
 right: `"numfmt: invalid number: ‘1x0K’"`', tests/common/util.rs:133:9

---- test_numfmt::test_should_report_invalid_empty_number_on_blank_stdin stdout ----
current_directory_resolved:
run: /home/drocco/sys/coreutils/target/debug/coreutils numfmt --from=auto
thread 'test_numfmt::test_should_report_invalid_empty_number_on_blank_stdin' panicked at 'assertion failed: `(left == right)`
  left: `"numfmt: error: Failed to parse suffix"`,
 right: `"numfmt: invalid number: ‘’"`', tests/common/util.rs:133:9

---- test_numfmt::test_should_report_invalid_suffix_on_stdin stdout ----
current_directory_resolved:
run: /home/drocco/sys/coreutils/target/debug/coreutils numfmt --from=auto
thread 'test_numfmt::test_should_report_invalid_suffix_on_stdin' panicked at 'assertion failed: `(left == right)`
  left: `"numfmt: error: Failed to parse suffix"`,
 right: `"numfmt: invalid suffix in input: ‘1k’"`', tests/common/util.rs:133:9

---- test_numfmt::test_should_skip_leading_space_from_stdin stdout ----
current_directory_resolved:
run: /home/drocco/sys/coreutils/target/debug/coreutils numfmt --from=auto
thread 'test_numfmt::test_should_skip_leading_space_from_stdin' panicked at 'assertion failed: `(left == right)`
  left: `""`,
 right: `"2048\n"`', tests/common/util.rs:119:9


failures:
    test_numfmt::test_leading_whitespace_in_free_argument_should_imply_padding
    test_numfmt::test_leading_whitespace_should_imply_padding
    test_numfmt::test_should_calculate_implicit_padding_per_free_argument
    test_numfmt::test_should_calculate_implicit_padding_per_line
    test_numfmt::test_should_convert_only_first_number_in_line
    test_numfmt::test_should_report_invalid_empty_number_on_blank_stdin
    test_numfmt::test_should_report_invalid_empty_number_on_empty_stdin
    test_numfmt::test_should_report_invalid_number_with_interior_junk
    test_numfmt::test_should_report_invalid_suffix_on_stdin
    test_numfmt::test_should_skip_leading_space_from_stdin

test result: FAILED. 16 passed; 10 failed; 0 ignored; 0 measured; 0 filtered out
```

Reply:
thanks :)
sorry for that, I missed the tests earlier, dunno why?!

Reply:
No problem! :)